cheeztest.php

```diff
@@ -1,364 +1,14 @@
-
-		if ( ( defined( 'DOING_CRON' ) && DOING_CRON )
-			|| ( defined( 'XMLRPC_REQUEST' ) && XMLRPC_REQUEST )
-			|| ( defined( 'WP_ADMIN' ) && WP_ADMIN )
-			|| is_admin()
-			|| $is_bot ) {
-			static::$is_excluded = true;
-			return false;
-		}
-		return true;
-	}
-
-	/**
-	 * write_segment_cookie - serves javascript response to client to write the user's
-	 * segment into a cookie so it will persist across visits
-	 *
-	 * @return
-	 */
-	function write_segment_cookie(){
-		?>
-			<script>
-				( function(){
-
-					// Check if cookies are enabled
-					// @see http://sveinbjorn.org/cookiecheck
-					var cookieEnabled = (navigator.cookieEnabled) ? true : false;
-
-					if (typeof navigator.cookieEnabled == "undefined" && !cookieEnabled){ 
-						document.cookie="testcookie";
-						cookieEnabled = (document.cookie.indexOf("testcookie") != -1) ? true : false;
-					}
-					
-					if( ! cookieEnabled )
-						return;
-					
-					// Set cookie with test condition
-					document.cookie = <?php echo json_encode( $this->name ); ?> + '=' + <?php echo json_encode( $this->group ); ?> + '; expires=' + <?php echo json_encode( $this->expires ); ?> + '; path=/';
-					
-				})();
-			</script>
-		<?php
-	}
-
-	/**
-	 * is_qualified_for - Helper func to get user's eligibility for any active test
-	 *
-	 * Returns bool true if user is qualified to see test - usage:
-	 *     if ( CheezTest::is_qualified_for( 'my-example-test' ) {
-	 *         //test-specific stuff goes here
-	 *     }
-	 *
-	 * @access	public
-	 * @param	string [ $name ] test name assigned during child initialization
-	 * @return  bool true if the user is eligible to participate (is qualified and has group)
-	 */
-	static function is_qualified_for( $name ){
-		if ( array_key_exists( $name, static::$active_tests ) ) {
-			return static::$active_tests[ $name ][ 'is_qualified' ] && static::$active_tests[ $name ][ 'group' ];
-		}
-		return false;
-	}
-
-	/**
-	 * get_group_for - Helper func to get user's group assignment for any active test
-	 *
-	 * Returns 'group' property of static $active_tests collection
-	 *
-	 * @access	public
-	 * @param	string [ $name ] test name assigned during child initialization
-	 * @return  mixed (string | bool) group name if user is eligible, false if not
-	 */
-	static function get_group_for( $name ){
-		if ( isset( static::$active_tests[ $name ] ) ) {
-			return static::$active_tests[ $name ][ 'group' ];
-		}
-		return false;
-	}
 
-	/**
-	 * is_in_group - Helper func to check if user is in the specified group
-	 *
-	 * @access	public
-	 * @param	string [ $name ] test name assigned during child initialization
-	 * @param	string [ $group ] group name to check against
-	 * @return	bool true if user is in in test group, false if not
-	 */
-	static function is_in_group( $name, $group ) {
-		return static::get_group_for( $name ) == $group;
-	}
-}
\ No newline at end of file
+if ( ! class_exists( 'CheezTest' ) ) {
```

Comment:
I tried avoiding the class_exists check, but without it I get stuck with the "Missing file doc comment (Squiz.Commenting.FileComment.Missing)" PHPCS feedback. Not sure if a bug in the sniff.